server/src-lib/Hasura/RQL/Types/ScheduledTrigger.hs

```diff
@@ -0,0 +1,110 @@
+-- | These are types for Scheduled Trigger definition; see "Hasura.Eventing.ScheduledTrigger"
+module Hasura.RQL.Types.ScheduledTrigger
+  ( ScheduleType(..)
+  , CreateScheduledTriggerWith(..)
+  , CreateScheduledTrigger(..)
+  , RetryConfST(..)
+  , formatTime'
+  ) where
+
+import           Data.Time.Clock
+import           Data.Time.Clock.Units
+import           Data.Time.Format
+import           Data.Aeson
+import           Data.Aeson.Casing
+import           Data.Aeson.TH
+import           Hasura.Prelude
+import           System.Cron.Types
+import           Hasura.Incremental
+
+import qualified Data.Text                     as T
+import qualified Data.Aeson                    as J
+import qualified Hasura.RQL.Types.EventTrigger as ET
+
+data RetryConfST
```

Comment:
@lexi-lambda When I use `DiffTime`, QuickCheck actually generates:

```
stRetryConf = RetryConfST {rcstNumRetries = 1, rcstIntervalSec = -0.092065381845s, rcstTimeoutSec = -2.083927801328s, rcstTolerance = -1.018299116354s}
```

Do you think this is okay? I would actually prefer if we only saw positive integers for `rcstIntervalSec` and `rcstTimeoutSec`.

Reply:
Turns out `Network.HTTP.Client.ResponseTimeout` is a type over `Int` so it also takes negative values so it might be okay.